src/qt/optionsmodel.cpp

```diff
@@ -403,13 +391,21 @@ void OptionsModel::SetPruneEnabled(bool prune, bool force)
     settings.setValue("bPrune", prune);
     const int64_t prune_target_mib = PruneGBtoMiB(settings.value("nPruneSize").toInt());
     std::string prune_val = prune ? ToString(prune_target_mib) : "0";
+    auto maybe_adjust_dash_options = [](){
+        if (gArgs.GetArg("-prune", 0) > 0) {
+            gArgs.SoftSetBoolArg("-disablegovernance", true);
+            gArgs.SoftSetBoolArg("-txindex", false);
+        }
+    };
     if (force) {
         gArgs.ForceSetArg("-prune", prune_val);
+        maybe_adjust_dash_options();
```

Comment:
If we're using `ForceSetArg` for `-prune`, should we extend that the param interactions?